extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -117,6 +124,24 @@ export class Bind {
      */
     this.digestQueuedAfterScan_ = false;
 
+    /**
+     * @const @private {!Array<Element>}
+     */
+    this.dynamicRoots_ = [];
+
+    this.subtreeMutationObserver_ = new MutationObserver(mutations => {
+      mutations.forEach(mutation => {
+        const mutatedNode = mutation.target;
+        if (this.dynamicRoots_.includes(mutatedNode)) {
+          this.removeBindingsForNode_(mutatedNode).then(() => {
+            return this.addBindingsForNode_(mutatedNode);
```

Comment:
Can we use `MutationRecord`'s `addedNodes` and `removedNodes` here rather than removing and rescanning the entire subtree?